src/forms/TextInput.jsx

```diff
@@ -71,7 +71,7 @@ const TextInput = (props) => {
 	// WC-158
 	// Only add a `value` prop if it is defined.
 	// Workaround for IE11 support (see ticket)
-	if (value) {
+	if (value || /^ *$/.test(value)) {
```

Comment:
`/^ *$/.test('asdf')` will return `false` with this regex.

We could instead just check that a string of any kind has been passed for `value`:

```
if (value || typeof value === 'string') {...}
```